core/arch/arm/mm/core_mmu.c

```diff
@@ -771,7 +734,6 @@ static void init_mem_map(struct tee_mmap_region *memory_map, size_t num_elems)
 				 &__start_phys_sdp_mem_section,
 				 &__end_phys_sdp_mem_section, "SDP");
 
-	check_sdp_intersection_with_nsec_ddr();
 #endif
```

Comment:
If the dyn SHM is discovered from `register_ddr()` (excluding some memories from an overall RAM range), the current routine should check that the platform did not register memories with using `register_nsec_ddr`.

I think this would be better for sanity: prevent platform from defining inconsistent informations.
Otherwise, we must clearly define the priorities between the `register_nsec_ddr`, `register_ddr` and the DTB parsing. In current implementation, DTB content overrides `register_nsec_ddr()` information.

Reply:
I will add add the check operation to see if the nsec_ddr had been register through `register_nsec_ddr()` before carve out memory.
If DT is enable, `carveout_memory_range()` will do nothing. And I prefer to put `carveout_memory_range()` before `init_fdt()`, so the the carve out memory can be overrides in the future.
So the priority will be `register_ddr < register_nsec_ddr < DTB parsing`.